ees_sharepoint/fetch_index.py

```diff
@@ -457,40 +458,113 @@ def index_permissions(
             groups.append(title)
         return groups
 
+    def index_sites(self, parent_site_url, ids, sites_path):
+        """ Indexes the site details to the Workplace Search
+            :param parent_site_url: parent site relative path
+            :param ids: id collection of the all the objects
+            :param sites_path: dictionary of site path and it's last updated time
+        """
+        _, datelist = get_partition_time(self.max_threads, self.start_time, self.end_time)
+        results = []
+        thread_pool = ThreadPool(self.max_threads)
+        for num in range(0, self.max_threads):
+            start_time_partition = datelist[num]
+            end_time_partition = datelist[num + 1]
+            thread = thread_pool.apply_async(
+                self.fetch_sites, (parent_site_url, {}, ids, (SITES in self.objects),
+                                   start_time_partition, end_time_partition))
+            results.append(thread)
+
+        sites, documents = [], []
+        for result in [r.get() for r in results]:
+            if result:
+                sites.append(result[0])
+                documents.extend(result[1])
+        thread_pool.close()
+        thread_pool.join()
+        self.threaded_index_documents(documents, SITES)
+        sites_path.extend(sites)
+
+    def index_lists(self, sites_path, ids, lists_details, libraries_details):
+        """ Indexes the list details to the Workplace Search
+            :param sites_path: dictionary of site path and it's last updated time
+            :param ids: id collection of the all the objects
+            :param lists_details: dictionary containing list name, list path and id
+            :param libraries_details: dictionary containing library name, library path and id
+        """
+        results = []
+        thread_pool = ThreadPool(self.max_threads)
+        partitioned_sites = partition_equal_share(sites_path, self.max_threads)
+        for site in partitioned_sites:
+            thread = thread_pool.apply_async(self.fetch_lists, (site, ids, (LISTS in self.objects)))
+            results.append(thread)
+        documents = []
+        for result in [r.get() for r in results]:
+            if result:
+                lists_details.update(result[0])
+                libraries_details.update(result[1])
+                documents.extend(result[2])
+        thread_pool.close()
+        thread_pool.join()
+        self.threaded_index_documents(documents, LISTS)
+
+    def index_items(self, job_type, lists_details, libraries_details, ids):
```

Comment:
I think it's worth to split this method into 2: 1 for list_items, another for drive_items. It'll make it easier to understand and handle - e.g. what if `job_type` passed is neither `list_items` nor `drive_items`, or situations like that.

Reply:
Sure, we can split this function into two functions: one is for list_item and another is for drive_items.